### PR TITLE
feat: add deploy

### PR DESCRIPTION
**Pull request by serge1peshcoff**
for this to work we need:

1) vars PRODUCTION_HOST, PRODUCTION_PORT, PRODUCTION_USER and PRODUCTION_PATH in .env
2) file secrets/ssh.key as a private key for SSH connection for user $PRODUCTION_USER
3) PATH_HUBOT to point at hubot-slack-docker/docker

hubot-troubleshooting is not a thing anymore apparently, so I've removed it from the list (it stopped node-ssh from being installed)

### CODE REVIEW

**Comment by linuxbandit** [comment]
thanks, I updated to a version which works reliably now (in short, no more EXTRA_things, all is through `external-packages.json`) 
can you cross check?

**Comment by serge1peshcoff** [comment]
@linuxbandit added one more commit to add newlines, other than that should be fine

**Comment by linuxbandit** [comment]
i think i found some sort of buggy behaviour (coming from the fact that nowhere says why should I put things in `external-scripts` vs `EXTRA_PACKAGES`). I am testing it, see if you manage to deploy something without my remedy (I may be wrong)

**Comment by serge1peshcoff** [comment]
I also don't get why do we need this boostrapping file with all that complexity vs just having a package.json in a root folder and having `npm install` in bootstrap.sh. (I was going to fix it later to not make this PR too big)

File: scripts/deploy.js [hunk 1]
```diff
@@ -1,3 +1,18 @@
+// Description:
+//   deploy script
+//
+// Dependencies:
+//   None
```
Comment (linuxbandit) [inline scripts/deploy.js:5]: depends on node-ssh and hubot-auth :D

File: docker/docker-compose.dev.yml [hunk 2]
```diff
@@ -6,4 +6,6 @@ services:
     image: aegee/hubot:dev
     volumes:
       - "./${PATH_HUBOT}/../scripts:/home/hubot/scripts:ro"
-
+    build:
```
Comment (linuxbandit) [inline docker/docker-compose.dev.yml:9]: can you put it on top of image?

Comment (serge1peshcoff) [reply docker/docker-compose.dev.yml:6]: done

File: scripts/deploy.js (hunk 1 above)
Comment (serge1peshcoff) [reply scripts/deploy.js]: agree, added it

**Review by linuxbandit — APPROVED** [review]
LGTM, and we saw it working. Only question, how come you `reply(\`` instead of `reply('` ? :D

**Comment by serge1peshcoff** [comment]
@linuxbandit answering your question:
image

**Pull request merged**